cpp/include/raft/neighbors/ivf_pq_types.hpp

```diff
@@ -104,6 +104,14 @@ struct index_params : ann::index_params {
    * flag to `true` if you prefer to use as little GPU memory for the database as possible.
    */
   bool conservative_memory_allocation = false;
+  /**
+   * The max number of data points to use per PQ code during PQ codebook training. Using more data
+   * points per PQ code may increase the quality of PQ codebook but may also increase the build
+   * time. The parameter is applied to both PQ codebook generation methods, i.e., PER_SUBSPACE and
+   * PER_CLUSTER. In both cases, we will use `pq_book_size * max_train_points_per_pq_code` training
+   * points to train each codebook.
+   */
+  uint32_t max_train_points_per_pq_code = 256;
```

Comment:
Why 256 here? Have we tested this empirically across many datasets ti verify this is a good default?

Reply:
The default value is inspired by FAISS, which also has 256 as default. We tested on  DEEP-100M here https://github.com/rapidsai/raft/pull/2052#issuecomment-1890191014. I will share results on other datasets.